### PR TITLE
Calculate badges in the new room list more reliably

### PR DESCRIPTION
**Pull request by turt2live**
Fixes https://github.com/vector-im/riot-web/issues/13757

This uses a different way of calculating the badges by using about 6 less variables, and consolidating the remaining ones down.

### CODE REVIEW

File: src/components/views/rooms/RoomTile2.tsx [hunk 1]
```diff
@@ -39,6 +39,14 @@ import * as FormattingUtils from "../../../utils/FormattingUtils";
  * warning disappears.                                             *
  *******************************************************************/
 
+enum NotificationColor {
+    // Inverted (None -> Red) because we do integer comparisons on this
+    None, // nothing special
```
Comment (t3chguy) [inline src/components/views/rooms/RoomTile2.tsx:44]: I think at least WebStorm doesn't like comments being after the entity, only on the line prior :(

Comment (turt2live) [reply src/components/views/rooms/RoomTile2.tsx:44]: image

Seems fine to me.

**Comment by t3chguy** [comment]
I mean if you ctrl Q it then it won't show up

**Comment by turt2live** [comment]
That feels like something to report to the WebStorm team and not fix in our project tbh. This isn't the only place we put the comments to the right, or even below, the definitions.

**Comment by t3chguy** [comment]
Fair enough, its buggy even when above:

image

shows the first comment for the third member.

**Pull request merged**